### DISCUSSION TITLE
spectrum similarity vlaues

### DISCUSSION THREAD

**Discussion by amrelguo** [vdemichev/DiaNN#717] [Q&A] [answered]
my understanding regarding the values in Spectrum.Similarity column in report.tsv file that it comes from matching the DIA spectrum to spectral library spectrum for certain peptide and it takes values from 0 to 1(1 means strong similarity). when I plotted the fragment ions from DDA spectral library against the fragment ions from DIA(in lib report file) for certain peptide"FDVNDEKNWGLSVYADKPETTKEQLGEF" , the match was perfect in terms of m/z and intensity values. so, I was expecting that its spectrum.similarity value will be 1 or very close to 1. however, it was calculated in report file as 8.2e-14. what is the reason?
thank you in advance.
![image](https://github.com/vdemichev/DiaNN/assets/133943092[report.log.txt](https://github.com/vdemichev/DiaNN/files/11608848/report.log.txt)826c7a10-6491-4d20-b2c8-d0c36b5c0dce)
[report_noML_best_1.csv](https://github.com/vdemichev/DiaNN/files/11608849/report_noML_best_1.csv)

**Comment by vdemichev** [comment, marked as answer]
Yes, the higher this value, the more similar the spectrum is to the library one. But I don't know if this measure is better than others like cosine, for example, this hasn't been tested. 

Perfect match means DIA-NN was not instructed to write experimental spectra to the library. But even if it was, would not be possible to infer similarity. For this, need to use --vis.

**Reply by amrelguo** [reply]
great. thanks alot!

**Reply by amrelguo** [reply]
but,  I am wondering if you mean with "Perfect match means DIA-NN was not instructed to write experimental spectra to the library." that it was happened accidentally or there is a certain parameter I have to consider to instruct DIA-NN to write the experimental spectra to the library. however, I checked generate spectral library checkbox.

**Reply by vdemichev** [reply]
'Library generation' setting

**Comment by amrelguo** [comment]
thanks so much
